### PR TITLE
Install from npm and jspm automatically after a pull/merge, if needed

### PR DESCRIPTION
**Pull request by sndrs**
better shell scripters than I might want to look over this (which I cobbled together from one I found via google)

@piuccio I couldn't find the `config.js` for facia tool? is it still knocking around?

cc @rich-nguyen @kelvin-chappell @mattandrews

### CODE REVIEW

**Comment by rich-nguyen** [comment]
I'd prefer to keep git hooks as light as possible. I think linting makes sense, but package management is a separate concern.

File: grunt-configs/shell.js [hunk 1]
```diff
@@ -34,10 +34,10 @@ module.exports = function(grunt, options) {
         },
 
         jspmInstallStatic: {
-            command: './node_modules/.bin/jspm install',
+            command: 'jspm clean && jspm install',
```
Comment (piuccio) [inline grunt-configs/shell.js]: doesn't this rely on a global jspm?

**Comment by piuccio** [comment]
[config.js for facia-tool](https://github.com/guardian/frontend/blob/master/facia-tool/public/js/jspm-config.js)

**Comment by mattandrews** [comment]
@rich-nguyen what if this script just alerted you that dependencies had changed and recommended you run the commands, rather than running them directly?

**Comment by sndrs** [comment]
> I'd prefer to keep git hooks as light as possible. I think linting makes sense, but package management is a separate concern.

thing is, many people work on this who don't really deal with asset management. the idea was just to make it invisible, in the way SBT just downloads packages when you run it (which seems useful to everyone).

why does this approach feel bad to you? 

I like @mattandrews idea - another option could be for `grunt hookmeup` to offer to add the asset installation `post-merge` script, rather than just doing it?

**Comment by rich-nguyen** [comment]
Isn't SBT more analogous to grunt, both are systems that produce executable output? I think there's a consistency that the SBT script downloads packages it needs, and grunt downloads the packages it needs (through `prepare`). git's operations don't require downloading of any packages.

I like the idea that a pull can warn you of new package dependencies, @mattandrews. I would then expect the user to run grunt to get them.

So my obvious question for @sndrs is do you mind having to run grunt after you pull from master?

**Comment by OliverJAsh** [comment]
Note: this won't work if you do `git pull --rebase`.

**Comment by sndrs** [comment]
> git's operations don't require downloading of any packages.

yeah that makes sense 

> So my obvious question for @sndrs is do you mind having to run grunt after you pull from master?

yeah that's totally fine with me - this was more aimed at trying to help people who had no idea why their branch had stopped working (or even that relevant the command existed). 

I'll update it to prompt with suggestions instead

**Comment by rich-nguyen** [comment]
sounds great, thanks @sndrs

**Comment by sndrs** [comment]
closing in favour of #9146

**Pull request closed without merging**